### DISCUSSION TITLE
Battery limit error (cant limit battery level and error pop up when clicking battery level)

### DISCUSSION THREAD

**Discussion by IamConfussion** [seerge/g-helper#2351] [Q&A]
![error](https://github.com/seerge/g-helper/assets/165627802/c4f15303-edee-46f2-ba8c-f6cff2903277)

Anybody know how to fix this? and also my laptop keep charging even after fully charge (battery health went from 96--93 in just 1 day)

**Comment by seerge** [comment]
@IamConfussion hello,

For battery limit please [check FAQ](https://github.com/seerge/g-helper/wiki/FAQ#battery-charge-limiter-is-not-working) . Often TUF devices don't support anything but fixed 80% as a limit, so try to set that.

As for error -  can you upload here everything from ``Details`` button in the alert?

**Reply by IamConfussion** [reply]
thanks for the reply. Ive tried everything in FAQ, still doesnt work. ive limit the batt to 80% too, still went past it

here's the detail

See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.ComponentModel.Win32Exception (2): An error occurred trying to start process 'powershell' with working directory 'C:\Windows\system32'. The system cannot find the file specified.
   at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
   at GHelper.Helpers.ProcessHelper.RunCMD(String name, String args) in D:\a\g-helper\g-helper\app\Helpers\ProcessHelper.cs:line 134
   at GHelper.SettingsForm.LabelCharge_Click(Object sender, EventArgs e) in D:\a\g-helper\g-helper\app\Settings.cs:line 269
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.Label.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, WM msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
System.Private.CoreLib
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Private.CoreLib.dll
----------------------------------------
GHelper
    Assembly Version: 0.164.0.0
    Win32 Version: n/a
    CodeBase: 
----------------------------------------
System.Runtime
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Runtime.dll
----------------------------------------
System.Windows.Forms
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6711
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.WindowsDesktop.App/7.0.16/System.Windows.Forms.dll
----------------------------------------
System.Threading.Thread
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Threading.Thread.dll
----------------------------------------
System.ComponentModel.Primitives
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.ComponentModel.Primitives.dll
----------------------------------------
System.Windows.Forms.Primitives
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6711
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.WindowsDesktop.App/7.0.16/System.Windows.Forms.Primitives.dll
----------------------------------------
System.Runtime.InteropServices
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Runtime.InteropServices.dll
----------------------------------------
System.Drawing.Primitives
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Drawing.Primitives.dll
----------------------------------------
System.Collections.Specialized
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Collections.Specialized.dll
----------------------------------------
Microsoft.Win32.SystemEvents
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.WindowsDesktop.App/7.0.16/Microsoft.Win32.SystemEvents.dll
----------------------------------------
System.Diagnostics.Process
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Diagnostics.Process.dll
----------------------------------------
System.Collections
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Collections.dll
----------------------------------------
System.ComponentModel.TypeConverter
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.ComponentModel.TypeConverter.dll
----------------------------------------
System.Text.Json
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Text.Json.dll
----------------------------------------
System.Threading
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Threading.dll
----------------------------------------
System.Collections.Concurrent
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Collections.Concurrent.dll
----------------------------------------
System.Private.Uri
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Private.Uri.dll
----------------------------------------
System.Text.Encodings.Web
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Text.Encodings.Web.dll
----------------------------------------
System.Reflection.Emit.Lightweight
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Reflection.Emit.Lightweight.dll
----------------------------------------
System.Reflection.Primitives
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Reflection.Primitives.dll
----------------------------------------
System.Reflection.Emit.ILGeneration
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Reflection.Emit.ILGeneration.dll
----------------------------------------
System.Memory
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Memory.dll
----------------------------------------
System.Text.Encoding.Extensions
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Text.Encoding.Extensions.dll
----------------------------------------
System.Numerics.Vectors
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Numerics.Vectors.dll
----------------------------------------
Microsoft.Win32.Primitives
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/Microsoft.Win32.Primitives.dll
----------------------------------------
System.Management
    Assembly Version: 7.0.0.1
    Win32 Version: n/a
    CodeBase: 
----------------------------------------
Microsoft.Win32.Registry
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/Microsoft.Win32.Registry.dll
----------------------------------------
System.Security.Principal.Windows
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Security.Principal.Windows.dll
----------------------------------------
System.Security.Claims
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Security.Claims.dll
----------------------------------------
System.Drawing.Common
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.WindowsDesktop.App/7.0.16/System.Drawing.Common.dll
----------------------------------------
System.Diagnostics.TraceSource
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Diagnostics.TraceSource.dll
----------------------------------------
System.Resources.Extensions
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.WindowsDesktop.App/7.0.16/System.Resources.Extensions.dll
----------------------------------------
System.Drawing
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6711
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.WindowsDesktop.App/7.0.16/System.Drawing.dll
----------------------------------------
NAudio.Wasapi
    Assembly Version: 2.1.0.0
    Win32 Version: n/a
    CodeBase: 
----------------------------------------
System.ComponentModel.EventBasedAsync
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.ComponentModel.EventBasedAsync.dll
----------------------------------------
Accessibility
    Assembly Version: 4.0.0.0
    Win32 Version: 7.0.1624.6711
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.WindowsDesktop.App/7.0.16/Accessibility.dll
----------------------------------------
System.ComponentModel
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.ComponentModel.dll
----------------------------------------
WinForms.DataVisualization
    Assembly Version: 1.0.0.0
    Win32 Version: n/a
    CodeBase: 
----------------------------------------
WinForms.DataVisualization.Utilities
    Assembly Version: 1.0.0.0
    Win32 Version: n/a
    CodeBase: 
----------------------------------------
netstandard
    Assembly Version: 2.1.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/netstandard.dll
----------------------------------------
HidSharpCore
    Assembly Version: 2.1.0.0
    Win32 Version: n/a
    CodeBase: 
----------------------------------------
System.Linq
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Linq.dll
----------------------------------------
System.Threading.ThreadPool
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Threading.ThreadPool.dll
----------------------------------------
Microsoft.Win32.TaskScheduler
    Assembly Version: 2.10.1.0
    Win32 Version: n/a
    CodeBase: 
----------------------------------------
System.ObjectModel
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.ObjectModel.dll
----------------------------------------
System.Diagnostics.FileVersionInfo
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Diagnostics.FileVersionInfo.dll
----------------------------------------
System.Text.RegularExpressions
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Text.RegularExpressions.dll
----------------------------------------
System.Xml.ReaderWriter
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Xml.ReaderWriter.dll
----------------------------------------
System.Private.Xml
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Private.Xml.dll
----------------------------------------
System.Runtime.Loader
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Runtime.Loader.dll
----------------------------------------
NvAPIWrapper
    Assembly Version: 0.8.1.100
    Win32 Version: n/a
    CodeBase: 
----------------------------------------
System.Diagnostics.PerformanceCounter
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.WindowsDesktop.App/7.0.16/System.Diagnostics.PerformanceCounter.dll
----------------------------------------
System.Threading.Overlapped
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Threading.Overlapped.dll
----------------------------------------
System.Runtime.Intrinsics
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Runtime.Intrinsics.dll
----------------------------------------
System.Collections.NonGeneric
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Collections.NonGeneric.dll
----------------------------------------
System.Net.Http
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Net.Http.dll
----------------------------------------
System.Net.Primitives
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Net.Primitives.dll
----------------------------------------
System.Diagnostics.Tracing
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Diagnostics.Tracing.dll
----------------------------------------
System.Diagnostics.DiagnosticSource
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Diagnostics.DiagnosticSource.dll
----------------------------------------
System.Net.Security
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Net.Security.dll
----------------------------------------
System.Security.Cryptography
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Security.Cryptography.dll
----------------------------------------
System.Net.Quic
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Net.Quic.dll
----------------------------------------
System.Net.Sockets
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Net.Sockets.dll
----------------------------------------
System.Net.NameResolution
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Net.NameResolution.dll
----------------------------------------
System.Diagnostics.StackTrace
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Diagnostics.StackTrace.dll
----------------------------------------
System.Reflection.Metadata
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Reflection.Metadata.dll
----------------------------------------
System.Collections.Immutable
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.Collections.Immutable.dll
----------------------------------------
System.IO.Compression
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1624.6629
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/7.0.16/System.IO.Compression.dll
----------------------------------------

************** JIT Debugging **************

**Reply by seerge** [reply]
@IamConfussion ok, thanks.

Error happened when you clicked on Battery Health label in the app that supposed to run powershell command to generate health report.

Apparently your system doesn't have powershell at all? :)

This build should prevent error from appearing at least
[GHelper.zip](https://github.com/seerge/g-helper/files/14823767/GHelper.zip)

Just in case it's not related to battery limit anyhow :)

**Reply by IamConfussion** [reply]
my system does have powershell, i think :')  all right, imma try that build
